cpu/stm32_common/periph/i2c_2.c

```diff
@@ -507,6 +507,7 @@ static inline int _wait_ready(I2C_TypeDef *i2c)
     return 0;
 }
 
+#if I2C_0_ISR || I2C_1_ISR
```

Comment:
Isn't better to use something like #ifdef MODULE_PERIPH_I2C instead ?

Reply:
Nope, this doesn't work:

```
$ make -C tests/driver_ads101x/ clean all BOARD=nucleo-f401re TOOLCHAIN=llvm
make: Entering directory '/home/mlenders/Repositories/RIOT-OS/RIOT/tests/driver_ads101x'
Building application "tests_driver_ads101x" for "nucleo-f401re" with MCU "stm32f4".

"make" -C /home/mlenders/Repositories/RIOT-OS/RIOT/boards/nucleo-f401re
"make" -C /home/mlenders/Repositories/RIOT-OS/RIOT/boards/common/nucleo
"make" -C /home/mlenders/Repositories/RIOT-OS/RIOT/core
"make" -C /home/mlenders/Repositories/RIOT-OS/RIOT/cpu/stm32f4
"make" -C /home/mlenders/Repositories/RIOT-OS/RIOT/cpu/cortexm_common
"make" -C /home/mlenders/Repositories/RIOT-OS/RIOT/cpu/cortexm_common/periph
"make" -C /home/mlenders/Repositories/RIOT-OS/RIOT/cpu/stm32_common
"make" -C /home/mlenders/Repositories/RIOT-OS/RIOT/cpu/stm32_common/periph
/home/mlenders/Repositories/RIOT-OS/RIOT/cpu/stm32_common/periph/i2c_2.c:511:20: error: unused function 'irq_handler' [-Werror,-Wunused-function]
static inline void irq_handler(i2c_t dev)
                   ^
1 error generated.
/home/mlenders/Repositories/RIOT-OS/RIOT/Makefile.base:83: recipe for target '/home/mlenders/Repositories/RIOT-OS/RIOT/tests/driver_ads101x/bin/nucleo-f401re/stm32_common_periph/i2c_2.o' failed
make[4]: *** [/home/mlenders/Repositories/RIOT-OS/RIOT/tests/driver_ads101x/bin/nucleo-f401re/stm32_common_periph/i2c_2.o] Error 1
/home/mlenders/Repositories/RIOT-OS/RIOT/Makefile.base:20: recipe for target 'ALL--/home/mlenders/Repositories/RIOT-OS/RIOT/cpu/stm32_common/periph' failed
make[3]: *** [ALL--/home/mlenders/Repositories/RIOT-OS/RIOT/cpu/stm32_common/periph] Error 2
/home/mlenders/Repositories/RIOT-OS/RIOT/Makefile.base:20: recipe for target 'ALL--/home/mlenders/Repositories/RIOT-OS/RIOT/cpu/stm32_common' failed
make[2]: *** [ALL--/home/mlenders/Repositories/RIOT-OS/RIOT/cpu/stm32_common] Error 2
/home/mlenders/Repositories/RIOT-OS/RIOT/Makefile.base:20: recipe for target 'ALL--/home/mlenders/Repositories/RIOT-OS/RIOT/cpu/stm32f4' failed
make[1]: *** [ALL--/home/mlenders/Repositories/RIOT-OS/RIOT/cpu/stm32f4] Error 2
/home/mlenders/Repositories/RIOT-OS/RIOT/tests/driver_ads101x/../../Makefile.include:414: recipe for target '/home/mlenders/Repositories/RIOT-OS/RIOT/tests/driver_ads101x/bin/nucleo-f401re/application_tests_driver_ads101x.a' failed
make: *** [/home/mlenders/Repositories/RIOT-OS/RIOT/tests/driver_ads101x/bin/nucleo-f401re/application_tests_driver_ads101x.a] Error 2
make: Leaving directory '/home/mlenders/Repositories/RIOT-OS/RIOT/tests/driver_ads101x'
```

Reply:
Also I think this is clearer than the `MODULE_PERIPH_I2C`, since the functions that call this function also are guarded by the respective macros in this expression.

Reply:
Alright then.